Src/Common/ActiveSubsciptionManager.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace Microsoft.ApplicationInsights.Common
+{
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.Linq;
+    using System.Threading;
+
+    /// <summary>
+    /// Keeps one active subscription to specific DiagnosticSource per process.
+    /// Helps manage subsciptions in scenarios where multiple apps hosted in the same process.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+#if DEPENDENCY_COLLECTOR
+    public
```

Comment:
why should this class be public?

Reply:
because we want to reuse it in the AspNetCore SDK for request deduplication

Reply:
Can we move this to the DependencyCollector project instead of Common?
The mixed accessibility levels seem wrong. :)